### ISSUE TITLE
Support Erlang/OTP 22

### ISSUE THREAD

**Issue by larshesel** [vernemq/vernemq#1196]
VerneMQ currently doesn not build on Erlang/OTP 22 due to some deprecation issues:

```
 $ make rel
cat vars.config > vars.generated
echo "{app_version, \"1.8.0-5-g7e458c6\"}." >> vars.generated
/home/mehdi/vernemq/rebar3  release
===> Verifying dependencies...
===> Compiling plumtree                                                         
===> Compiling _build/default/lib/plumtree/src/plumtree_metadata_exchange_fsm.erl failed
_build/default/lib/plumtree/src/plumtree_metadata_exchange_fsm.erl:79: gen_fsm:start/3 is deprecated and will be removed in a future release; use gen_statem:start/3
_build/default/lib/plumtree/src/plumtree_metadata_exchange_fsm.erl:86: gen_fsm:send_event/2 is deprecated and will be removed in a future release; use gen_statem:cast/2
_build/default/lib/plumtree/src/plumtree_metadata_exchange_fsm.erl:303: gen_fsm:send_event/2 is deprecated and will be removed in a future release; use gen_statem:cast/2

make: *** [Makefile:28: rel] Error 1 
```

We should fix this.

Reported in vernemq/plumtree#23

**Comment by codeadict** [comment]
Could be closed too

**Comment by larshesel** [comment]
Closed via #1213
